### PR TITLE
fix some warnings

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -274,7 +274,15 @@ object Settings {
   )
 
   def slickScalacSettings = Seq(
-    scalacOptions ++= List("-deprecation", "-feature", "-unchecked", "-Xfuture"),
+    scalacOptions ++= List("-deprecation", "-feature", "-unchecked"),
+    scalacOptions ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, v)) if v <= 12 =>
+          Seq("-Xfuture")
```
Comment (WellingR) [inline project/Settings.scala]: Wasn't `-Xfuture` deprecated in 2.12 already?
Also do we need this scalac option at all for 2.11? I am curious what feature we are using which needs this flag...

File: slick/src/main/scala/slick/ast/Type.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ trait AtomicType extends Type {
 }
 
 final case class StructType(elements: ConstArray[(TermSymbol, Type)]) extends Type {
-  override def toString = "{" + elements.iterator.map{ case (s, t) => s + ": " + t }.mkString(", ") + "}"
+  override def toString = "{" + elements.iterator.map{ case (s, t) => s.toString + ": " + t }.mkString(", ") + "}"
```
Comment (WellingR) [inline slick/src/main/scala/slick/ast/Type.scala]: You can also use an interpolator here. I believe it does slightly better optimization for the string concatenation

File: slick/src/main/scala/slick/ast/Type.scala [hunk 3]
```diff
@@ -128,7 +128,7 @@ final case class ProductType(elements: ConstArray[Type]) extends Type {
 }
 
 final case class CollectionType(cons: CollectionTypeConstructor, elementType: Type) extends Type {
-  override def toString = cons + "[" + elementType + "]"
+  override def toString = cons.toString + "[" + elementType + "]"
```
Comment (WellingR) [inline slick/src/main/scala/slick/ast/Type.scala]: Use a string interpolator

**Comment by WellingR** [comment]
Looks good! I am definitely in favour of fixing the sea of warnings which is being logged for scala 2.12 and 2.13.

**Comment by xuwei-k** [comment]
rebased

File: project/Settings.scala (hunk 1 above)
Comment (SethTisue) [reply project/Settings.scala:276]: Someone could dig into this in a separate PR.

**Comment by SethTisue** [comment]
thank you!

**Pull request merged**